c++/src/kj/async-io-test.c++

```diff
@@ -300,10 +300,10 @@ TEST(AsyncIo, AncillaryMessageHandlerNoMsg) {
 #endif
 
 // We only support ancillary messages on Unix.
-// MacOS only supports SO_TIMESTAMP on datagram sockets, so this test won't work.
+// MacOS & FreeBSD only support SO_TIMESTAMP on datagram sockets, so this test won't work.
 // Android fails this test for unknown reasons and I don't care because it's an extremely obscure
 // feature anyway.
-#if !_WIN32 && !__CYGWIN__  && !__APPLE__ && !__ANDROID__
+#if !_WIN32 && !__CYGWIN__  && !__APPLE__ && !__ANDROID__ && !__FreeBSD__
```

Comment:
Actually looking at this again, if we're literally filtering every platform except linux, maybe we should just make it `#if __linux__ && !__ANDROID__`?

```suggestion
// SO_TIMESTAMP specifically seems to have lots of portability issues, so we only even attempt this
// test on Linux. Even Android fails this test for some unknown reason, so we avoid it there. This
// is an extremely obscure feature and it's just not worth the time to debug the test on every
// platform.
#if __linux__ && !__ANDROID__
```

Reply:
That comment is somewhat misleading in my view, I must point out. There is nothing wrong with SO_TIMESTAMP support (it's been around way before Linux came along), the only problem is that the test case specifically written around using that feature with SOCK_STREAM, a (mis-)feature that has been added into Linux kernel around 2009. I even dug a commit message at some point, but is too lazy to find it out. The whole idea of timestamping a data on a stream connection has no much sense, since you might be reading data from multiple TCP packets in the same recvmsg() call.

Reply:
P.S. The correct way to fix this would be to re-write test case to use datagram sockets, IMHO. (not volunteering) :)

Reply:
Sure, I didn't write this test and wasn't aware of the specific details. I believe the test is specifically trying to test ancillary messages on streams, though, so changing it to test datagram sockets would defeat the purpose. How about:

```
// This test uses SO_TIMESTAMP on a SOCK_STREAM, which is only supported by Linux. Ideally we'd
// rewrite the test to use some other message type that is widely supported on streams. But for
// now we just limit the test to Linux. Also, it doesn't work on Android for some reason, and it
// isn't worth investigating, so we skip it there.
```